File: runTests.cpp
#include<stdio.h>
#include<unistd.h>

int main () {
    pid_t pid[20];
    int i, index;
    FILE *fPointer = popen("ls -al", "r");
    if (fPointer == 0) {
        fprintf(stderr, "Could not execute\n");
    }
    for (i = 0; i < 20; i++) {
        index = i;
        if (index == i) {

            //pid[index] = fork();
            printf ("i am %d\n", index);
        }
    }
    // java -classpath bin havabol.HavaBol for_loop.txt
    system("java -classpath 
    return 0;    
}

File: runTests.py
#!/usr/bin/python

import os
import sys
import time
from subprocess import call
# external command to run havabol
# java -classpath bin havabol.HavaBol for_loop.txt
pids = []

testFiles = os.listdir(sys.argv[1])
outputFiles = []
for testFile in testFiles:
    outputFiles.append(testFile + "Output")
for i in range (0, len(testFiles)):
    index = i
    if (index == i):
        pids.append(index)
        pids[index] = os.fork()
        if (pids[index] == 0):
            time.sleep(i)
            call(["java", "-classpath", "bin", "havabol.HavaBol", sys.argv[1] + testFiles[index]])
            print ("Running on file " + testFiles[index])
            sys.exit(0)
